src/Storage/Sha1FileHashingService.php

```diff
@@ -178,7 +178,11 @@ private function getCache()
     private function buildCacheKey($fileID, $fs)
     {
         $fsID = $this->getFilesystemKey($fs);
-        return base64_encode(sprintf('%s://%s', $fsID, $fileID));
+        $cacheKey = base64_encode(sprintf('%s://%s', $fsID, $fileID));
+        // base64_encode can contain `/` , but that is not a valid character in cache keys
+        // @see CacheItem::validateKey. Therefore we translate it to `a` (effectively a random character)
+        // without actually being random since the key needs to be consistent
+        return strtr($cacheKey, ['/' => 'a']);
```

Comment:
Should we just preg_replace all the reserved characters `{}()/\@:` while we’re here?

Reply:
From what I can see `base64_encode` will use the characters outlined in [rfc4648](https://tools.ietf.org/html/rfc4648#page-6) of which are two outside of alpha-numeric values: `+` and `/`, and `+` is not forbidden.

Happy to err on the side of caution though and add them all in, did we want to assign each character a letter or just replace all instances with the same character?

Reply:
Why not use the base64 URL encoding standard rather than replacing / with an a (which has a base64 meaning)

Reply:
Oh good point, I should've scrolled further down in the RFC, updated it to use `_`